server/components/users/user-model.js

```diff
@@ -0,0 +1,33 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+
+const userSchema = new mongoose.Schema({
+	name: {
+		type: String,
+		required: true
+	},
+	email: {
+		type: String,
+		required: true
+	},
+	passwordHash: {
+		type: String,
+		required: true
+	}
+});
+
+userSchema.pre("save", next => {
+	try {
+		const hash = bcrypt.hash(this.passwordHash, 10);
```

Comment:
I'm not sure where we stand regarding authentication for the moment being since we may move to Firebase Auth for authentication, but I'll review this nonetheless. I think we should hash the password on the client side and then send the **hash** to the server. Another possibility is to hash the password once on the frontend and once on the backend.

Reply:
@R-Layer It would be great if you could reply to this.